### ISSUE TITLE
TOC not updating correctly in Chrome

### ISSUE THREAD

**Issue by BartJanvanRossum** [r-lib/pkgdown#1981]
The content menu is not unfolding as expected. More specifically, selecting the bottom item in the content list unfolds the item above and selects that item and within the item the last  subitem. For https://github.com/Biometris/statgenSTA/tree/develop:

![image](https://user-images.githubusercontent.com/28725410/147738512-d0a7eead-2e0a-4cf3-be5c-57fe4cfb08dc.png)

The page does still move to the correct location and manually scrolling further down, even just one line, will unfold the correct item in the content menu:

![image](https://user-images.githubusercontent.com/28725410/147738640-c9c3f84e-4727-494a-9dcc-cb464abe45b7.png)

I tested this with both the CRAN version and the main github branch of pkgdown with identical results. Also there is no difference whether bootstrap 5 is used or not.

**Comment by hadley** [comment]
Can you please provide a link to the rendered page too?

**Comment by BartJanvanRossum** [comment]
Sure:
https://bartjanvanrossum.github.io/statgenSTA/articles/statgenSTA.html

**Comment by hadley** [comment]
Unfortunately I can't see what the problem is. Do you might writing a itemised list of the actions you're taking (e.g. click here, then click there) and what you expect?

**Comment by BartJanvanRossum** [comment]
In the content menu clicking on an item does three things:
- scroll to that particular item in the document
- unfold the item revealings its sub items in the content menu
- highlight the item in the content menu 


In the vignette linked above it works fine here
- click on "Creating a TD object" in the content menu
    - scroll to that particular item in the document ✔️ 
    - unfold the item revealings its sub items in the content menu ✔️ 
    - highlight the item in the content menu ✔️ 

    
However it does not here (screenshot in original issue)
- click on "Extract model results" in the content menu
    - scroll to that particular item in the document ✔️ 
    - unfold the item revealings its sub items in the content menu ❌
        - the menu item above is unfolded    
    - highlight the item in the content menu ❌ 
        - the last sub item in the sub menu of the item above in the content menu is highlighted
 - scroll down in the document (1 line is enough)
     - the correct content menu item is now unfolded and highlighted
        

While writing this down I tested with a few different browsers. It seems the behavior on Edge and Chrome is worse than on Firefox. On Firefox it is only the last menu item that doesn't work as expected. On both Edge and Chrome only the 1st and 3rd item in the content menu work as expected when clicked.

**Comment by hadley** [comment]
Can you please provide a link to the vignette that doesn't work?

**Comment by BartJanvanRossum** [comment]
Yes, but it is exactly the same link from a few comments up:
https://bartjanvanrossum.github.io/statgenSTA/articles/statgenSTA.html

**Comment by hadley** [comment]
Does it work or not? I am very confused.

**Comment by BartJanvanRossum** [comment]
I thought I was very clear, but obviously I wasn't. To reproduce the problem

- open https://bartjanvanrossum.github.io/statgenSTA/articles/statgenSTA.html
- in the content menu, click "Extracting model results"

Result:
- scrolls to section
- unfolds "Modeling" in content menu
- highlights "Modeling" and "Model reports" in content menu

Expected:
- scrolls to section
- unfolds "Extracting model results" in content menu
- highlights "Extracting model results" in content menu

So in short, the wrong section is unfolded and the wrong section is highlighted.

**Comment by hadley** [comment]
Ok the source of the confusion is that it works fine for me in Safari, but not in chrome.

**Comment by hadley** [comment]
Even more minimal reprex: visit https://bartjanvanrossum.github.io/statgenSTA/articles/statgenSTA.html#extracting-model-results in chrome. "Model reports" is highlighted instead of "Extracting model results"

Probably something to do with `scroll-padding`.

**Comment by hadley** [comment]
But the problem doesn't seem to occur on https://pkgdown.r-lib.org/articles/customise.html#sidebar

**Comment by BartJanvanRossum** [comment]
The difference might be that "Extracting model results" is the last section in the TOC. In {pkgdown} the sidebar section is somewhere in the middle of the TOC. In {statgenSTA} highlighting works fine when clicking e.g. "Creating a TD object", a section with subsections higher up in the TOC.

**Comment by hadley** [comment]
One possible issue to look into is that I should now be supplying `rootMargin` instead of `offset`: https://getbootstrap.com/docs/5.3/components/scrollspy/#options

**Comment by hadley** [comment]
I think this is beyond my ability to debug — I think someone with deeper knowledge of js will need to read the source code of the scrollspy plugin to figure out what's going wrong 😞
